creation/views.py

```diff
@@ -254,40 +278,67 @@ def creation_reject_role_request(request, recid):
+def creation_revoke_role_request(request, role_type,langid):
     roles = {
         'contributor': 0,
         'external-contributor': 1,
         'video-reviewer': 2,
         'domain-reviewer': 3,
         'quality-reviewer': 4,
     }
-    if role_type in roles:
-        try:
-            role_rec = RoleRequest.objects.get(user = request.user, role_type = roles[role_type], status = 1)
-            if role_rec.role_type != 2:
-                if role_rec.role_type == 0 or role_rec.role_type == 1:
-                    ContributorRole.objects.filter(user = role_rec.user).update(status = 0)
-                elif role_rec.role_type == 3:
-                    DomainReviewerRole.objects.filter(user = role_rec.user).update(status = 0)
-                elif role_rec.role_type == 4:
-                    QualityReviewerRole.objects.filter(user = role_rec.user).update(status = 0)
-                role_rec.user.groups.remove(Group.objects.get(name = role_type.title()))
-                role_rec.status = 2
-                role_rec.save()
-                messages.success(request, role_type.title() + ' role has been revoked from ' + role_rec.user.username)
-        except:
-            raise PermissionDenied()
-    else:
-        messages.error(request, 'Invalid role type argument!')
+    
+    group_role_id = {
+            0: 'Contributor',
+            1: 'External-Contributor',
+            2: 'Video-Reviewer',
+            3: 'Domain-Reviewer',
+            4: 'Quality-Reviewer',
+        }
+
+    # Revoke multiple languages from the user
+    languages = langid.split('/')
+
+    for a_language in languages:
+
+        if role_type in roles:
+            try:
+                role_rec = RoleRequest.objects.get(user = request.user, role_type = roles[role_type], status = 1,language_id = a_language)
+                
+                
+                if role_rec.role_type != 2:
+                    if role_rec.role_type == 0 or role_rec.role_type == 1:
+                        ContributorRole.objects.filter(user = role_rec.user).update(status = 0)
+                                                    
+                    elif role_rec.role_type == 3:
+                        DomainReviewerRole.objects.filter(user = role_rec.user).update(status = 0)
+                    elif role_rec.role_type == 4:
+                        QualityReviewerRole.objects.filter(user = role_rec.user).update(status = 0)
+                    rec_check = RoleRequest.objects.filter(user = request.user, role_type = roles[role_type],status=1)                    
+                    role_rec.status = 2
+                    role_rec.save()
+                
+                    lang_show = Language.objects.get(id = a_language)
+                    messages.success(request, role_type.title() + ' role has been revoked from ' + role_rec.user.username+' for the language '+ lang_show.name)
+
+                    
+            except:
```

Comment:
Not good as exception is not stated!
Also if exception then PermissionDenied is called. Why an exception will occur?